### DISCUSSION TITLE
Guice error after removing akka

### DISCUSSION THREAD

**Discussion by chetrebman** [playframework/playframework#13015] [General]
After upgrading to 3.0.5 from 2.8.8 I get the following guice injector error, even though I believe
I have made all changes to remove akka from the system.
```
java.lang.NoClassDefFoundError: akka/stream/Materializer
...
com.google.inject.Guice.createInjector(Guice.java:78)
play.api.inject.guice.GuiceBuilder.injector(GuiceInjectorBuilder.scala:200)
play.api.inject.guice.GuiceApplicationBuilder.build(GuiceApplicationBuilder.scala:176)
play.api.inject.guice.GuiceApplicationLoader.load(GuiceApplicationLoader.scala:21)
play.core.server.DevServerStart$DevServerApplicationProvider$1.reload(DevServerStart.scala:225)
play.core.server.DevServerStart$DevServerApplicationProvider$1.get(DevServerStart.scala:190)
play.core.server.PekkoHttpServer.handleRequest(PekkoHttpServer.scala:320)
```

**Comment by wsargent** [comment]
Play 3.0 uses Pekko, not Akka.  If you have an Akka reference, it's because the project (or a transitive dependency) added it.

The easiest way to debug this is to use JDWP with `suspend=y` and attach to the debug process https://stackoverflow.com/questions/24218341/how-to-run-play-framework-2-x-in-debug-mode-in-intellij-idea

And then when you click play in the debug process in IntelliJ IDEA, it will take you to the breakpoint in the stack where that exception is thrown and you'll be able to see what arguments and code led up to that point through step debugging.

**Comment by chetrebman** [comment]
Thanks, for the response but you clearly didn't read the problem. I followed the migration guide to remove akka references from previous version and I still get the error. I do an sbt dependancy tree and see no mention of akka. THAT is the problem

________________________________

**Reply by chetrebman** [reply]
I spoke to soon, I did find this in the sbt dependency tree, but I don't know what it means lol
```
| +-org.playframework:play-ws_2.13:3.0.5 [S]
  | | +-akka:pekko-stream_2.13:1.0.3 [S]
```

**Reply by wsargent** [reply]
I read the problem fine.  Your problem is that an akka reference exists -- Play 3.0 uses Pekko internally, so it cannot be a problem of Play itself.  It must be either a direct dependency or a transitive dependency in your project.

**Comment by chetrebman** [comment]
We are in complete agreement. I also do not believe it is the play framework which I have been upgrading, without complaint, since version 1.6

I also  firmly believe there is an akka reference. I was hoping to get some insight into how to find it, since I have spent hours looking for it. I will look again tomorrow. If I find it, I will be sure to share my insights for future travelers.

________________________________

**Comment by chetrebman** [comment]
I finally found it. I forgot that I had created a /lib directory many eons ago that had an old mailer which must have used the Actor system somehow. Once the jar was removed the system booted. :) Not sure why I didn't see it in the dependency tree, but at this point I just going to move on lol
